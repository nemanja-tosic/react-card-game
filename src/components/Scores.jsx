// @flow
import React from 'react';

import { Game } from '../models/game';
import './Scores.css';

const Scores = ({game, onNewGame}: { game: Game, onNewGame: () => void }) => (
  <div className="Scores container-fluid">
    <div className="row align-items-center justify-content-center">
      <div className="col-md-6 col-lg-4">
        {game.players.map((player, index) => (
          <div key={index} className={`Scores__score${game.winners.includes(player) ? ' Scores__score--winner' : ''}`}>
            <div className="Scores__name">
              {player.name} {game.winners.includes(player) && ' won!!!'}
            </div>
            <div className="Scores__total">{player.score}</div>
          </div>
        ))}
        <button type="button" className="btn btn-primary" onClick={() => onNewGame()}>
          Back to the menu
        </button>
      </div>
    </div>
  </div>
);

export default Scores;
