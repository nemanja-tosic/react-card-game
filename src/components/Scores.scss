.Scores {
  height: 100%;
}

.Scores, .Scores > .row {
  height: 100%;
}

.Scores__name {
  flex: 1;
}

.Scores__score {
  margin-bottom: 1rem;
  padding: 0.7rem 0;
  display: flex;
}

.Scores__score--winner {
  font-weight: bold;
  color: gold;
}

.Scores__score:last-child {
  margin-bottom: 0;
}
