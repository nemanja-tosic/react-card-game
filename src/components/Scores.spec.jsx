// @flow
import React from 'react';
import { shallow } from 'enzyme';

import { TestGame } from '../test/support/test-game';
import Scores from './Scores';
import { TestPlayer } from '../test/support/test-player';

describe('Scores', () => {
  let component;
  let onNewGameSpy;
  let game;

  beforeEach(() => {
    game = TestGame.create({
      players: [
        TestPlayer.create({name: 'John'}),
        TestPlayer.create({name: 'HAL'}),
      ]
    });
    onNewGameSpy = jest.fn();
    component = shallow(<Scores game={game} onNewGame={onNewGameSpy}/>);
  });

  describe('when there is a single winner', () => {
    it('should indicate the winner', () => {
      game.winners = [game.players[0]];

      component.setProps({game});

      expect(component.getNode()).toMatchSnapshot();
    });
  });

  describe('when there are multiple winners', () => {
    it('should highlight all the winners', () => {
      game.winners = [...game.players];

      component.setProps({game});

      expect(component.getNode()).toMatchSnapshot();
    });
  });

  it('should call the onNewGame prop when the new game button is clicked', () => {
    component.find('button').simulate('click');

    expect(onNewGameSpy).toHaveBeenCalledTimes(1);
  });
});
