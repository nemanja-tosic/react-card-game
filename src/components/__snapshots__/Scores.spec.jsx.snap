// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scores when there are multiple winners should highlight all the winners 1`] = `
<div
  className="Scores container-fluid"
>
  <div
    className="row align-items-center justify-content-center"
  >
    <div
      className="col-md-6 col-lg-4"
    >
      <div
        className="Scores__score Scores__score--winner"
      >
        <div
          className="Scores__name"
        >
          John
           
           won!!!
        </div>
        <div
          className="Scores__total"
        />
      </div>
      <div
        className="Scores__score Scores__score--winner"
      >
        <div
          className="Scores__name"
        >
          HAL
           
           won!!!
        </div>
        <div
          className="Scores__total"
        />
      </div>
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Back to the menu
      </button>
    </div>
  </div>
</div>
`;

exports[`Scores when there is a single winner should indicate the winner 1`] = `
<div
  className="Scores container-fluid"
>
  <div
    className="row align-items-center justify-content-center"
  >
    <div
      className="col-md-6 col-lg-4"
    >
      <div
        className="Scores__score Scores__score--winner"
      >
        <div
          className="Scores__name"
        >
          John
           
           won!!!
        </div>
        <div
          className="Scores__total"
        />
      </div>
      <div
        className="Scores__score"
      >
        <div
          className="Scores__name"
        >
          HAL
           
        </div>
        <div
          className="Scores__total"
        />
      </div>
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Back to the menu
      </button>
    </div>
  </div>
</div>
`;
